pallets/loans/src/lib.rs

```diff
@@ -187,12 +186,12 @@ pub mod pallet {
 	#[pallet::storage]
 	#[pallet::getter(fn get_next_loan_id)]
 	pub(crate) type NextLoanId<T: Config> =
-		StorageMap<_, Blake2_128Concat, PoolIdOf<T>, u128, ValueQuery, OnNextLoanIdEmpty>;
+		StorageMap<_, Blake2_128Concat, PoolIdOf<T>, u128, ValueQuery, OnNextLoanIdEmpty>; // Could it be ValueStorage?
```

Comment:
I was thinking if `LoanId` is unique, maybe we don't need to store the "next" loan Id per pool, is enough with only one `NextLoanId` used for all pools.